homeassistant/components/blueprint/websocket_api.py

```diff
@@ -96,25 +104,26 @@ async def ws_import_blueprint(hass, connection, msg):
     }
 )
 @websocket_api.async_response
-async def ws_save_blueprint(hass, connection, msg):
+async def ws_save_blueprint(
+    hass: HomeAssistant,
+    connection: websocket_api.ActiveConnection,
+    msg: dict[str, Any],
+) -> None:
     """Save a blueprint."""
 
     path = msg["path"]
     domain = msg["domain"]
 
-    domain_blueprints: dict[str, models.DomainBlueprints] | None = hass.data.get(
-        DOMAIN, {}
-    )
+    domain_blueprints: dict[str, models.DomainBlueprints] = hass.data.get(DOMAIN, {})
 
     if domain not in domain_blueprints:
         connection.send_error(
             msg["id"], websocket_api.ERR_INVALID_FORMAT, "Unsupported domain"
         )
 
     try:
-        blueprint = models.Blueprint(
-            yaml.parse_yaml(msg["yaml"]), expected_domain=domain
-        )
+        yaml_data = cast(dict[str, Any], yaml.parse_yaml(msg["yaml"]))
```

Comment:
I originally adjusted this to use `assert`, but I figured it was better to leave the validation up to `Blueprint`.